### PR TITLE
Remove dead bundle code [#178284429]

### PR DESCRIPTION
**Pull request by dougmartin**
- Removed all controllers, helpers, policies and specs for the dataservice bundle
- Added migration to remove all tables and columns related to the removed code

### CODE REVIEW

**Comment by dougmartin** [comment]
@scytacki Noah and I think you should pack some sandwiches before this review.

We feel pretty confident in all the changes - we *tried* to be very methodical.  Tests are green!

**Comment by dougmartin** [comment]
@scytacki / @knowuh while reviewing our notes from yesterday when starting the next story I realized we left the last step off - the migration to remove the unneeded tables and columns.  I'll add that now.

**Comment by dougmartin** [comment]
I've added the migration to remove the models/columns. (https://github.com/concord-consortium/rigse/pull/1024/commits/94452fd295f2386ee9495b4c2e26179c1e31c6fb)

**Comment by dougmartin** [comment]
@scytacki - we do have one last task:

- figure out alternative for learner#calculate_last_run (did use various bundle updated_at times)

we switched it to use `last_run` in this PR.  Thoughts?

**Comment by dougmartin** [comment]
I missed one table when removing the `bundle_content_id` column so I added a commit that updates that migration: https://github.com/concord-consortium/rigse/pull/1024/commits/c896db9bdc6f9a69da9cecd25e2815bbed196e02

**Review by scytacki — CHANGES_REQUESTED** [review]
Overall this looks great!

I found a couple of minor issues. And one possibly big one around the 'last_run'. I'm being lazy and not doing the experimentation myself to confirm the difference between LARA runtime and AP assignments. So I'm asking you to do it.

File: rails/app/models/report/learner.rb [hunk 1]
```diff
@@ -71,30 +71,8 @@ def self.build_last_run_string(last_run, opts={})
...
-      bucket_time = bucket_logger.bucket_contents.last.updated_at
-    end
-
-    times = [pub_time,bundle_time,bucket_time].compact.sort
-    if times.size > 0
-      self.last_run = times.last
-    end
-
+    # RAILS6 TODO: figure out alternative? - this was using the now removed dataservice model update times
+    return self.last_run
```
[the remark this answers is not in the document]
Comment (knowuh) [reply rails/app/models/report/learner.rb:75]: Your theory is correct, I verified it on staging.

I ran two activites (AP and Standard) a few times.  The standard activity updated the `report_learner.last_run` whereas the AP did not.  Details:

### 2021-06-21 1:00 PM:

```ruby
	irb(main):023:0> ls.each { |l| puts [l.first.offering.name, l.first.report_learner.last_run].join  }
	AP Smoke Test 2021-06-21 15:06:08 UTC
	LARA Smoke Test Seq2021-06-21 15:11:26 UTC
```

### 2021-06-21 1:14 PM (after running both and waiting a bit_:

```ruby
	irb(main):032:0> ls.each { |l| puts [l.first.offering.name, l.first.report_learner.last_run].join(" :: ")  }
	AP Smoke Test  :: 2021-06-21 15:06:08 UTC
	LARA Smoke Test Seq :: 2021-06-21 17:04:25 UTC
```

[the remark this answers is not in the document]
Comment (knowuh) [reply rails/app/models/report/learner.rb:75]: I am going to follow step *`ii.`change the code* in your recommendations above, and maybe swing back round to hunting down what dataservice model is being updated, and documenting that as a second pass.

**Comment by knowuh** [comment]
OK I think we are ready for another review.  

Because this is a big PR, I think we only need to re-review the changes requested  ( aab79bb → 3a9e6d6 )

Because I did it without @dougmartin maybe he should review along with @scytacki 

## master merge and rebase:

Rails6 had master merged into it on 2021-06-21.  This PR wasn't merged yet, so I rebased it on the updated Rails6 for ease of review.

## Addressing PR requests:

### Remove unused run_status and launch routes & actions:
- https://github.com/concord-consortium/rigse/pull/1024/commits/aab79bb8a7f2eb0f433ac24ef648f127e51180e4
- https://github.com/concord-consortium/rigse/pull/1024/commits/4139d83b8c9647ef873624738af3d6e6c99bf429
- https://github.com/concord-consortium/rigse/pull/1024/commits/ab2e2679f4e090b5ceadd9d430ee6b3432c34146

### Re-add the learners to eager fetch of the active_users query:
- https://github.com/concord-consortium/rigse/pull/1024/commits/b6b593d9b2d6988d6fb7a84b24af65f1df68979b

### Fixes to `last_run`:
- https://github.com/concord-consortium/rigse/pull/1024/commits/9c23cdbfa93cd1789409458a1be94d6c97b66da7
- https://github.com/concord-consortium/rigse/pull/1024/commits/3a9e6d6795cc77f63ef65cd360df26330a95e0b0

**Comment by dougmartin** [comment]
New commits look good :+1:

**Review by scytacki — APPROVED** [review]
Looks great to me.

**Pull request merged**